File: app/src/main/java/com/techsolutions/appgrade/View/FirstActivity.java
package com.techsolutions.appgrade.View;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.techsolutions.appgrade.R;
import com.techsolutions.appgrade.database.FireBaseDBUtils;

/**
 * Created by Udi on 6/6/2017.
 * Activity for unregister user.
 * Let him decide if he want to navigate to start register in normal way or just find partner fast
 */

public class FirstActivity extends Activity {

    private Button btnRegister;
    private Button btnFindPartner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.first_activity);

        btnRegister = (Button) findViewById(R.id.btnBuildProfile);
        btnFindPartner = (Button) findViewById(R.id.btnFindFast);

        // User pressed on find partner fast button
        btnFindPartner.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent findFast = new Intent("com.techsolutions.appgrade.SEMESTERACTIVITY");
                startActivity(findFast);
            }
        });

        // User pressed on create profile button
        btnRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent registerIntent = new Intent("com.techsolutions.appgrade.REGISTERACTIVITY");
                startActivity(registerIntent);
            }
        });

        FireBaseDBUtils.checkVersion();
    }
}

File: app/src/main/java/com/techsolutions/appgrade/View/FirstActivity.kt
package com.techsolutions.appgrade.View

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.first_activity.*
import android.widget.Button
import com.techsolutions.appgrade.R

class FirstActivity : AppCompatActivity() {

    // UI Component
    private lateinit var btnFindPartner: Button
    private lateinit var btnRegister: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.first_activity)

        btnFindPartner = findViewById(R.id.btnFindFast) as Button
        btnRegister = findViewById(R.id.btnBuildProfile) as Button

        // User pressed on find partner fast button
        btnFindPartner.setOnClickListener {
            val findFast = Intent("com.techsolutions.appgrade.SEMESTERACTIVITY")
            startActivity(findFast)
        }

        // User pressed on create profile button
        btnRegister.setOnClickListener {
            val registerIntent = Intent("com.techsolutions.appgrade.REGISTER")
            registerIntent.putExtra("CurrentStatus", "User_NotRegister")
            startActivity(registerIntent)
        }
    }
}
